File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/social/ContactLookMessage.scala





















// Generated on 05/03/2013 10:35:00
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ContactLookMessage extends MessageDeserializer {
  val messageId = 5934

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ContactLookMessage.messageId

    val requestId: Int
    val playerName: String
    val playerId: Int
    val look: EntityLook.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeInt(requestId)
      buf.writeString(playerName)
      buf.writeInt(playerId)
      look.serialize(buf)
    }

  }

  protected def create(requestId: Int, playerName: String, playerId: Int, look: EntityLook.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val requestId = buf.readInt()
    val playerName = buf.readString()
    val playerId = buf.readInt()
    val look = EntityLook.deserialize(buf)

    create(requestId, playerName, playerId, look)
  }

}

object ContactLookMessage extends ContactLookMessage {
  case class MessageImpl(requestId: Int, playerName: String, playerId: Int, look: EntityLook.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(requestId: Int, playerName: String, playerId: Int, look: EntityLook.TypeImpl) = MessageImpl(requestId, playerName, playerId, look)

  def apply(requestId: Int, playerName: String, playerId: Int, look: EntityLook.TypeImpl) = create(requestId, playerName, playerId, look)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ContactLookMessage.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ContactLookMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5934;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int requestId;
    public String playerName;
    public int playerId;
    public EntityLook look;
    
    public ContactLookMessage() { }
    
    public ContactLookMessage(int requestId, String playerName, int playerId, EntityLook look) {
        this.requestId = requestId;
        this.playerName = playerName;
        this.playerId = playerId;
        this.look = look;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(requestId);
        buf.writeString(playerName);
        buf.writeInt(playerId);
        look.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        requestId = buf.readInt();
        if (requestId < 0)
            throw new RuntimeException("Forbidden value on requestId = " + requestId + ", it doesn't respect the following condition : requestId < 0");
        playerName = buf.readString();
        playerId = buf.readInt();
        if (playerId < 0)
            throw new RuntimeException("Forbidden value on playerId = " + playerId + ", it doesn't respect the following condition : playerId < 0");
        look = new EntityLook();
        look.deserialize(buf);
    }
    
}
